src/cargo/core/compiler/mod.rs

```diff
@@ -1166,6 +1172,62 @@ fn features_args(unit: &Unit) -> Vec<OsString> {
     args
 }
 
+/// Generates the `--remap-path-scope` and `--remap-path-prefix` for [RFC 3127].
+/// See also unstable feature [`-Ztrim-paths`].
+///
+/// [RFC 3127]: https://rust-lang.github.io/rfcs/3127-trim-paths.html
+/// [`-Ztrim-paths`]: https://doc.rust-lang.org/nightly/cargo/reference/unstable.html#profile-trim-paths-option
+fn trim_paths_args(
+    cmd: &mut ProcessBuilder,
+    cx: &Context<'_, '_>,
+    unit: &Unit,
+    scopes: &[ProfileTrimPaths],
+) -> CargoResult<()> {
+    if scopes.iter().all(|s| matches!(s, ProfileTrimPaths::None)) {
```

Comment:
Shouldn't this be `any` (instead of `all`)?
Otherwise you will end-up with the invalid scope: `none` from `ProfileTrimPaths::as_str`

Reply:
Nice catch! Fixed.